### ISSUE TITLE
Mandatory Fields for Family Records

### ISSUE THREAD

**Issue by joesmare** [mareinc/mare#309]
THIS STORY evolved as part of another issue. Per @n-devr we shouldn't implement this, at this time, as a mandatory field fix to the data. We will try and clean-up records as a matter of procedure and protocol. We can revisit in future. 

n-devr commented 23 hours ago
As I'm coding this solution I'm noticing that we have to put guards and define 'smart' defaults in the case that a family doesn't have a preferred age range set. We could avoid this and implement a cleaner, more consistent, solution by making those fields required in the Family model, or by defining default values for them. Is that something that we could look into?

@joesmare @bchiaruttini

 @joesmare
 
Member
joesmare commented 21 hours ago
@n-devr your suggestion makes sense, with a caveat with regard to when it is made mandatory. We could make the necessary fields mandatory...but only at the point where the 'Registered with MARE' box is checked and prevent a save immediately after that action is taken. This would include the following fields:

Current Children in Family: Number of Children
Matching Preferences: From Age
Matching Preferences: To Age
Matching Preferences: Minimum Number of Children
Matching Preferences: Maximum Number of Children
This extends beyond your initial questions, as related to age only, but these are all fields required for reporting and to avoid null values.

 @n-devr
 
Member
Author
n-devr commented 20 hours ago
OK - given the difficulty of making fields required only after certain conditions have been met I think we should try to tackle this with smart defaults, or perhaps it's not worth changing at all. Having to contend with unset or missing values will add complexity to the queries made from the reporting dashboard, but that may be a necessary evil.

We should discuss in more detail when we meet next. It may be worth having a meeting about this with a wider group as well, as the decision we make here will impact functionality across reporting dashboards.

n-devr added a commit that referenced this issue 19 hours ago
 @n-devr
Fix age range search on Child Matching dashboard (#302)
5d4b0af
@n-devr
 
Member
Author
n-devr commented 18 hours ago
I ended up writing a query that could handle the missing fields, let's just touch base to ensure there aren't other reasons to move towards implementing smart defaults for the fields you listed. It's certainly still an option and would make the query logic simpler and improve the display of the reporting dashboard, however it would also introduce risk to any exisitng code that isn't expecting default values.

**Comment by bchiaruttini** [comment]
does not appear to continue to be an issue
